middleware/auth.js

```diff
@@ -0,0 +1,10 @@
+const authMiddleware = (req, res, next) => {
```

Comment:
Nitpick:
This can be simplified using an early return:
```js
const authMiddleware = (req, res, next) => {
  if (req.user) return next();

  req.flash("error", "You can't access that page before logon.");
  res.redirect("/");
}
```